apis/v1alpha1/collector_webhook.go

```diff
@@ -360,11 +401,31 @@ func checkAutoscalerSpec(autoscaler *AutoscalerSpec) error {
 	return nil
 }
 
-func SetupCollectorWebhook(mgr ctrl.Manager, cfg config.Config) error {
+// warningsGroupedByResource is a helper to take the missing permissions and format them as warnings.
+func warningsGroupedByResource(reviews []*v1.SubjectAccessReview) []string {
+	fullResourceToVerbs := make(map[string][]string)
+	for _, review := range reviews {
+		if review.Spec.ResourceAttributes != nil {
+			key := fmt.Sprintf("%s/%s", review.Spec.ResourceAttributes.Group, review.Spec.ResourceAttributes.Resource)
```

Comment:
nit: If group is empty, I think we should just use the resource name without the leading slash.